### PR TITLE
Add empty dataloader CLI and core sub projects

### PR DESCRIPTION
**Pull request by ypeckstadt**
## Description

To start the migration of moving the Data Loader code to the main repository, we first need to have two new subprojects in the repository.  A data loader CLI and core subproject. This PR adds these subprojects.

## Related issues and/or PRs

NA

## Changes made

- Added data loader folder with sub folders for the CLI and core subprojects
- Added both projects to the settings.gradle file
- Added the based code for the DataLoaderCLI tool with two empty commands: Import and Export
- Added the build.gradle files for each project.

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes.
- [ ] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [ ] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

I have not added the maven publish and signature code yet in the build.gradle files. I assume it will be a copy of the schema-loader build.gradle file. However, for now, it's better to not add it yet to avoid any publish risks. Will add it via another PR in the future.

## Release notes

NA

### CODE REVIEW

File: data-loader/cli/build.gradle
Comment (Torch3333) [inline data-loader/cli/build.gradle:53]: It's minor but can you add an empty line here?

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!
I just left a minor comment.

**Review by brfrn169 — APPROVED** [review]
LGTM other than what @Torch3333 pointed out. Thank you!

**Comment by ypeckstadt** [comment]
@brfrn169 It seems some of the tests are failing but does not really seem related to the content of my PR.  Is there something specific I need to be carefull of when pushing content to this repo to avoid this? (unless it actually is caused by my added changes of course.)

**Comment by brfrn169** [comment]
@ypeckstadt I re-ran the failed tests, and they were successful. It looks like it's a timing issue.

Should we review this again?

**Comment by ypeckstadt** [comment]
@brfrn169 Thank you. I think it's ok to merge. The only new thing I added, after the talk on Slack, is the build.gradle file in the dataloader folder that defines the group name, avoiding the issues with the duplicate`core` named subprojects when including them.

**Comment by ypeckstadt** [comment]
@brfrn169 Just to double check, would it be ok for me to merge after everything is approved and the checks pass? Or better to leave that up to you or Vincent?

**Comment by brfrn169** [comment]
@ypeckstadt You can merge after everything is approved and the checks pass.

**Pull request merged**
